File: Editor/ICSharpCode.AvalonEdit/bin/Debug/Templates-Data/ProjectTemplates/Web/VisualBasic/1033/WebApplication40/Account/OpenAuthProviders.ascx.vb
Imports Microsoft.AspNet.Membership.OpenAuth

Partial Class Account_OpenAuthProviders
    Inherits System.Web.UI.UserControl

    Protected Sub Page_Load(ByVal sender As Object, ByVal e As System.EventArgs) Handles Me.Load
        $if$ ($targetframeworkversion$ == 4.0)
        AddHandler Page.PreRenderComplete, AddressOf Page_PreRenderComplete

        $endif$
        If IsPostBack Then
            Dim provider = Request.Form("provider")
            If provider Is Nothing Then
                Return
            End If

            Dim redirectUrl = "~/Account/RegisterExternalLogin"
            If Not String.IsNullOrEmpty(ReturnUrl) Then
                Dim resolvedReturnUrl = ResolveUrl(ReturnUrl)
                redirectUrl += "?ReturnUrl=" + HttpUtility.UrlEncode(resolvedReturnUrl)
            End If

            OpenAuth.RequestAuthentication(provider, redirectUrl)
        End If
    End Sub

    $if$ ($targetframeworkversion$ == 4.0)
    Protected Sub Page_PreRenderComplete(ByVal sender As Object, ByVal e As System.EventArgs)
        providersList.DataSource = OpenAuth.AuthenticationClients.GetAll()
        providersList.DataBind()
    End Sub

    Protected Function Item(Of T As Class)() As T
        Return TryCast(Page.GetDataItem(), T)
    End Function
    $endif$

    Public Property ReturnUrl As String

    $if$ ($targetframeworkversion$ >= 4.5)
    Public Function GetProviderNames() As IEnumerable(Of ProviderDetails)
        Return OpenAuth.AuthenticationClients.GetAll()
    End Function
    $endif$
End Class

File: VSExplorer/bin/Debug/s/Templates-Data/ProjectTemplates/Web/CSharp/1033/WebApplication40/Account/OpenAuthProviders.ascx.cs
using System;
using System.Collections.Generic;
using System.Web;
using Microsoft.AspNet.Membership.OpenAuth;

public partial class Account_OpenAuthProviders : System.Web.UI.UserControl
{
    protected void Page_Load(object sender, EventArgs e)
        {
            $if$ ($targetframeworkversion$ == 4.0)
            Page.PreRenderComplete += Page_PreRenderComplete;

            $endif$
            if (IsPostBack)
            {
                var provider = Request.Form["provider"];
                if (provider == null)
                {
                    return;
                }
                
                var redirectUrl = "~/Account/RegisterExternalLogin";
                if (!String.IsNullOrEmpty(ReturnUrl))
                {
                    var resolvedReturnUrl = ResolveUrl(ReturnUrl);
                    redirectUrl += "?ReturnUrl=" + HttpUtility.UrlEncode(resolvedReturnUrl);
                }
                
                OpenAuth.RequestAuthentication(provider, redirectUrl);
            }
        }

        $if$ ($targetframeworkversion$ == 4.0)
        protected void Page_PreRenderComplete(object sender, EventArgs e)
        {
            providersList.DataSource = OpenAuth.AuthenticationClients.GetAll();
            providersList.DataBind();
        }

        protected T Item<T>() where T : class
        {
            return Page.GetDataItem() as T ?? default(T);
        }
        $endif$

        public string ReturnUrl { get; set; }

        $if$ ($targetframeworkversion$ >= 4.5)
        public IEnumerable<ProviderDetails> GetProviderNames()
        {
            return OpenAuth.AuthenticationClients.GetAll();
        }
        $endif$
}